internal/app/query/cache.go

```diff
@@ -0,0 +1,235 @@
+package query
```

Comment:
License header is missing, please copy from another file.

Reply:
Might be more safe for the reviewers and version control to move this code back where it resided.